src/test/ProposalPayloadE2E.t.sol

```diff
@@ -3,37 +3,335 @@ pragma solidity ^0.8.15;
+    IReserveInterestRateStrategyV2 public constant OLD_INTEREST_RATE_STRATEGY_ETHEREUM =
+        IReserveInterestRateStrategyV2(0xfC0Eace19AA7498e0f36eF1607D282a8d6debbDd);
+
+    // New Strategies
+    address public constant NEW_INTEREST_RATE_STRATEGY_ETHEREUM = 0x04c28D6fE897859153eA753f986cc249Bf064f71;
+    address public constant NEW_INTEREST_RATE_STRATEGY_POLYGON = 0x80cb7e9E015C5331bF34e06de62443d070FD6654;
+    address public constant NEW_INTEREST_RATE_STRATEGY_POLYGON_V3 = 0x4b8D3277d49E114C8F2D6E0B2eD310e29226fe16;
+
+    // Util Addresses
+    address public constant CROSSCHAIN_FORWARDER_POLYGON = 0x158a6bC04F0828318821baE797f50B0A1299d45b;
+    address public constant BRIDGE_ADMIN = 0x0000000000000000000000000000000000001001;
+    address public constant FX_CHILD_ADDRESS = 0x8397259c983751DAf40400790063935a11afa28a;
+    address public constant POLYGON_BRIDGE_EXECUTOR = 0xdc9A35B16DB4e126cFeDC41322b3a36454B1F772;
+
+    // Underlying
+    address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+    address public constant BAL_POLYGON = 0x9a71012B13CA4d3D0Cdc72A177DF3ef03b0E76A3;
+
+    string internal ETHEREUM = AaveAddressBookV2.AaveV2Ethereum;
+    string internal POLYGON = AaveAddressBookV2.AaveV2Polygon;
+
+    // New Strategies
+    IDefaultInterestRateStrategy public constant strategyPolygonV3 =
+        IDefaultInterestRateStrategy(NEW_INTEREST_RATE_STRATEGY_POLYGON_V3);
+    IReserveInterestRateStrategyV2 strategy = IReserveInterestRateStrategyV2(NEW_INTEREST_RATE_STRATEGY_ETHEREUM);
+    IReserveInterestRateStrategyV2 strategyPolygon = IReserveInterestRateStrategyV2(NEW_INTEREST_RATE_STRATEGY_POLYGON);
 
     function setUp() public {
         // To fork at a specific block: vm.createSelectFork(vm.rpcUrl("mainnet"), BLOCK_NUMBER);
-        vm.createSelectFork(vm.rpcUrl("mainnet"));
+        mainnetFork = vm.createFork(vm.rpcUrl("mainnet"), 16435723);
+        polygonFork = vm.createFork(vm.rpcUrl("polygon"), 38278604);
 
-        // Deploy Payload
-        ProposalPayload proposalPayload = new ProposalPayload();
+        // Deploy Payloads
+        vm.selectFork(polygonFork);
+        proposalPayloadPolygon = new ProposalPayloadPolygon();
+
+        vm.selectFork(mainnetFork);
+        proposalPayload = new ProposalPayload();
 
         // Create Proposal
         vm.prank(GovHelpers.AAVE_WHALE);
-        proposalId = DeployMainnetProposal._deployMainnetProposal(
+        proposalId = DeployProposals._deployMainnetProposal(
             address(proposalPayload),
+            address(proposalPayloadPolygon),
             0x344d3181f08b3186228b93bac0005a3a961238164b8b06cbb5f0428a9180b8a7 // TODO: Replace with actual IPFS Hash
         );
     }
 
+    // Utility to transform memory to calldata so array range access is available
+    function _cutBytes(bytes calldata input) public pure returns (bytes calldata) {
+        return input[64:];
+    }
+
     function testExecute() public {
```

Comment:
Would recommend splitting apart the tests for the Ethereum and Polygon sections. Would make it a lot more clearer than handling it all in one test.